src/series.jl

```diff
@@ -14,7 +14,10 @@ const SeriesData = Union{AVec{<:DataPoint}, Function, Surface, Volume}
 
 prepareSeriesData(x) = error("Cannot convert $(typeof(x)) to series data for plotting")
 prepareSeriesData(::Nothing) = nothing
-prepareSeriesData(s::SeriesData) = handlemissings(s)
+prepareSeriesData(s::SeriesData) = handleinfinites(handlemissings(s))
+
+handleinfinites(s) = s
+handleinfinites(s::AbstractArray{<:MaybeNumber}) = [isinf(x) ? NaN : x for x in s]
```

Comment:
This copies the input data. We already do that in one place, in they `computexy` functions. Could we handle that there so we don't copy twice?
Alternatively, given that we do copy the input data, it should be safe to use `replace!` here.